### PR TITLE
Opal 157 removable accounts

### PR DESCRIPTION
**Pull request by norvarju**

### CODE REVIEW

File: app/src/main/java/com/greenfox/kalendaryo/adapter/AccountAdapter.java [hunk 1]
```diff
@@ -48,6 +66,51 @@ public void setEmailChange(EmailChange emailChange) {
 
     @Override
     public void onBindViewHolder(AccountAdapter.ViewHolder holder, int position) {
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                AlertDialog.Builder alert = new AlertDialog.Builder(view.getContext());
+                alert.setTitle("Warning!");
+                alert.setMessage("Are you sure to delete account?");
+                alert.setPositiveButton("YES", new DialogInterface.OnClickListener() {
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/AccountAdapter.java]: Use 'Yes'

File: app/src/main/java/com/greenfox/kalendaryo/adapter/AccountAdapter.java [hunk 2]
```diff
@@ -48,6 +66,51 @@ public void setEmailChange(EmailChange emailChange) {
...
+
+                            @Override
+                            public void onFailure(Call<Void> call, Throwable t) {
+                                Toast.makeText(view.getContext(),"Account can not be deleted", Toast.LENGTH_LONG).show();
+                            }
+                        });
+                        dialog.dismiss();
+                    }
+                });
+                alert.setNegativeButton("NO", new DialogInterface.OnClickListener() {
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/adapter/AccountAdapter.java]: Use 'No' or 'Cancel'

File: app/src/main/java/com/greenfox/kalendaryo/ChooseAccountActivity.java [hunk 3]
```diff
@@ -84,24 +87,35 @@ public void onFailure(Call<PostKalendarResponse> call, Throwable t) {
...
         accountNamesView.setLayoutManager(recyclerLayoutManager);
         DividerItemDecoration dividerItemDecoration =
                 new DividerItemDecoration(accountNamesView.getContext(),
                         recyclerLayoutManager.getOrientation());
         accountNamesView.addItemDecoration(dividerItemDecoration);
+    }
 
-        AccountAdapter accountAdapter = new
-                AccountAdapter(kalpref.getGoogleAuths(),this);
+    public void getAuthResponse(String clientToken) {
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/ChooseAccountActivity.java]: rename AuthResponse

File: app/src/main/java/com/greenfox/kalendaryo/fragments/AccountsFragment.java [hunk 4]
```diff
@@ -5,37 +5,51 @@
...
         kalpref = new KalPref(getActivity());
-        adapter = new AccountsList(view.getContext());
-        accountNamesView = view.findViewById(R.id.allaccounts);
-        accountNamesView.setAdapter(adapter);
+        DaggerApiComponent.builder().build().inject(this);
+        recyclerView = view.findViewById(R.id.accountRecyclerView);
+        LinearLayoutManager recyclerLayoutManager = new LinearLayoutManager(this.getContext());
+        recyclerView.setLayoutManager(recyclerLayoutManager);
         floatingActionButton = view.findViewById(R.id.addNewAccount);
+        getAuthResponse(kalpref.clientToken());
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/fragments/AccountsFragment.java]: getAuthResponse() logic is added here and in `ChooseAccountActivity.java` too
maybe could refactor and move the logic into a service?

File: app/src/main/java/com/greenfox/kalendaryo/ChooseAccountActivity.java [hunk 5]
```diff
@@ -84,24 +87,35 @@ public void onFailure(Call<PostKalendarResponse> call, Throwable t) {
...
+                });
+                accountNamesView.setAdapter(adapter);
+            }
 
-        accountAdapter.setEmailChange(new AccountAdapter.EmailChange() {
             @Override
-            public void emailChanged(String email) {
-                kalendar.setOutputGoogleAuthId(email);
+            public void onFailure(Call<AuthResponse> call, Throwable t) {
+
```
Comment (arnikaeva) [inline app/src/main/java/com/greenfox/kalendaryo/ChooseAccountActivity.java]: do not leave empty failure blocks, print out the error messages

**Review by arnikaeva — CHANGES_REQUESTED** [review]
Nice work :) only some refactoring needed

**Review by pataricza1987 — DISMISSED** [review]

**Review by arnikaeva — APPROVED** [review]

**Review by pataricza1987 — APPROVED** [review]

**Review by bekobarna — APPROVED** [review]

**Pull request merged**